doc/rm/c_cpp_coding_style.md

```diff
@@ -186,6 +186,37 @@ This may change if we find cases where this initialization improves readability.
 When initializing an array, initializers *may* be designated when that makes the array more readable (e.g., lookup tables that are mostly zeroed). Mixing designated and undesignated initializers, or using nested initializers, is still
 forbidden.
 
+### Inline Functions
+
+Functions that we strongly wish to be inlined, and which are part of a public interface, should be declared as an inline function.
+This annotation serves as an indication to the programmer that the function has a low calling overhead, despite being part of a public interface.
+Presence---or lack---of an `inline` annotation does not imply a function will---or will not---be inlined by the compiler.
+
+C11 standardised inline functions, learning from the mistakes in C++ and various nonstandard extensions.
+This means there are many legacy ways to define an inline function in C.
+We have chosen to follow how C11 designed the `inline` keyword.
+
+The function definition is written in a header file, with the keyword `inline`:
+```c
+// in my_inline.h
+inline int my_inline_function(long param1) {
+  // implementation
+}
+```
+
+There should be exactly one compilation unit with a compatible `extern` declaration of the same function:
+```c
+// in my_inline.c
+#include <my_inline.h>
+extern int my_inline_function(long param1);
+```
+
+Any compilation unit that includes `my_inline.h` should be linked to the compilation unit with the extern declarations.
+This ensures that if the compiler chooses not to inline `my_inline_function`, there is a function definition that can be called.
+
+Functions marked `static` should not be marked `inline`.
+The compiler is capable of inlining static functions without the `inline` annotation, and they should not be part of any public interface.
```

Comment:
Maybe we should call this out separately as well: "static linkage qualification is forbidden in header files, since this will cause the symbol to be replicated in every translation unit including it."

Reply:
I've added this, and moved all things addressing `static` (including `static inline`) into a section only about Static Declarations